docs/user/adding-new-metrics.md

```diff
@@ -6,54 +6,61 @@ The format of that file is documented [here](https://mozilla.github.io/glean_par
 
 When adding a new metric, the workflow is:
 
-* Decide on which [metric type](metrics/index.md) you want to use.
+* Consider the question you are trying to answer with telemetry, and choose the [metric type](metrics/index.md) and parameters to use.
 * Add a new entry to [`metrics.yaml`](#The-metricsyaml-file).
 * Add code to your project to record into the metric by calling the Glean SDK.
 
 > **Important**: Any new data collection requires documentation and data-review.
 This is also required for any new metric automatically collected by the Glean SDK.
 
-## The `metrics.yaml` file
+## Choosing a metric type
 
-The [`metrics.yaml` file](https://mozilla.github.io/glean_parser/metrics-yaml.html) defines the metrics your application or library will send.
-They are organized into categories.
-The overall organization is:
+### Is it a simple (scalar) measurement?
 
-```YAML
-# Required to indicate this is a `metrics.yaml` file
-$schema: moz://mozilla.org/schemas/glean/metrics/1-0-0
+If the value is true or false, use a [boolean metric](metrics/boolean.html).
 
-toolbar:
-  click:
-    type: event
-    description: |
-      Event to record toolbar clicks.
-    notification_emails:
-      - CHANGE-ME@example.com
-    bugs:
-      - 123456789
-    data_reviews:
-      - http://example.com/path/to/data-review
-    expires: 2019-06-01  # <-- Update to a date in the future
+If the value is a string, use a [string metric](metrics/string.html).  If you want to store multiple strings in a group, use a [string list metric](metrics/string_list.html).
 
-  double_click:
-    ...
+If you need to record a closely-related group of strings, or you don't know the things the strings are related to at build time, use a [labeled string metric](metrics/labled_strings.html).
 
-category2.subcategory:  # Categories can contain subcategories
-  metric:
-    ...
+For all of these simple value types, you should consider how the value changes in the application and how that relates to the ping it is being sent in. When the ping is submitted, it will contain the "last known" value for the metric, and then the value is reset according to its [lifetime](#What-is-the-lifetime-of-the-data).
 
-```
+If you need to know the current state of a value, such as a user preference, to provide context for other metrics, be aware that the lifetime of the value may not be the same as the lifetime of the ping it is in. You can address this either with a custom ping that matches the lifetime of the value, or by using a user lifetime metric so the value will persist across pings.
```

Comment:
Aside from just being hard to explain:

I think for metrics that want to record the status of a preference (like the environment stuff on desktop) so that it can be related to other metrics, I think the best recommendation in our toolbox is a user lifetime metric, but I know others have *opinions* about that.

Reply:
> I think the best recommendation in our toolbox is a user lifetime metric, but I know others have _opinions_ about that.

:-P 

I personally think that the status of a preference (exactly like the environment on desktop) is `application` lifetime: the _specific_ environment on Desktop is built at startup (ok, in a staged way) and lives throughout the life of Firefox (or as long as it is changed).

A relevant example, for desktop environment, is search defaults: it's set [later at startup](https://searchfox.org/mozilla-central/rev/91f6c02fcf4c16f78fdc4417f61f192688294066/toolkit/components/telemetry/app/TelemetryEnvironment.jsm#1418) and, in the rare circumstance of pings being generated before that, it will not be present in pings. Still, that's not a "user lifetime": it lives as long as the application lives.

I think the core of the issue here is understanding what it's being measured and when/how it should be available: that's indeed an hard task :)

Reply:
Maybe expanding on the gfx metric example of the `compositor` would be a good way to illustrate the dangers of using application lifetime metrics with the metrics ping, and that it requires some extra care? That seemed like a good real-world case that shows where application lifetime metrics can be weird.

Reply:
We can leave it as a warning that the lifetime might not match and let the rest of the documentation suggest solutions appropriate to the reader. Or we could be a bit more precise about "If it is important that every ping contain some value for this metric, it should have a 'ping' lifetime and be sent on a custom ping you control."